components/opc-ua/src/Microsoft.Azure.IIoT.OpcUa.Edge.Publisher/src/Engine/MonitoredItemMessageEncoder.cs

```diff
@@ -88,7 +97,6 @@ public class MonitoredItemMessageEncoder : IMessageEncoder {
             var current = notifications.GetEnumerator();
             var processing = current.MoveNext();
             var messageSize = 2; // array brackets
-            maxMessageSize -= 2048; // reserve 2k for header
```

Comment:
why deleted this line? we need to reserve space for header in payload

Reply:
I moved it to a constant _encodedMessageSizeOverhead in DataFlowProcessingEngine.cs. The payload is [protocol-agnostic](https://docs.microsoft.com/en-us/azure/iot-hub/iot-hub-devguide-messages-construct#message-size), so we shouldn't need to reserve so much space for the header.